x-pack/plugins/spaces/public/share_saved_objects_to_space/share_saved_objects_to_space_column.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiBadge } from '@elastic/eui';
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { EuiToolTip } from '@elastic/eui';
+import {
+  SavedObjectsManagementColumn,
+  SavedObjectsManagementRecord,
+} from '../../../../../src/plugins/saved_objects_management/public';
+import { SpaceTarget } from './types';
+import { SpacesManager } from '../spaces_manager';
+
+type SpaceMap = Map<string, SpaceTarget>;
+
+export class ShareToSpaceSavedObjectsManagementColumn extends SavedObjectsManagementColumn<
+  SpaceMap
+> {
+  public id: string = 'share_saved_objects_to_space';
+
+  public euiColumn = {
+    field: 'namespaces',
+    name: i18n.translate('xpack.spaces.management.shareToSpace.columnTitle', {
+      defaultMessage: 'Shared spaces',
+    }),
+    description: i18n.translate('xpack.spaces.management.shareToSpace.columnDescription', {
+      defaultMessage: 'The other spaces that this object is currently shared to',
+    }),
+    render: (namespaces: string[] | undefined, object: SavedObjectsManagementRecord) => {
+      if (!this.data) {
+        return null;
+      }
+
+      const authorized = namespaces?.filter((namespace) => namespace !== '?') ?? [];
+      const authorizedSpaceTargets: SpaceTarget[] = [];
+      authorized.forEach((namespace) => {
+        const spaceTarget = this.data!.get(namespace);
+        if (spaceTarget === undefined) {
+          // in the event that a new space was created after this page has loaded, fall back to displaying the space ID
+          authorizedSpaceTargets.push({
+            id: namespace,
+            name: namespace,
+            disabledFeatures: [],
+            isActiveSpace: false,
+          });
+        } else if (!spaceTarget.isActiveSpace) {
+          authorizedSpaceTargets.push(spaceTarget);
+        }
+      });
+      const unauthorizedCount = (namespaces?.filter((namespace) => namespace === '?') ?? []).length;
+      const unauthorizedTooltip = i18n.translate(
+        'xpack.spaces.management.shareToSpace.columnUnauthorizedLabel',
+        { defaultMessage: 'You do not have permission to view these spaces' }
+      );
+
+      return (
+        <EuiFlexGroup wrap responsive={false} gutterSize="xs">
+          {authorizedSpaceTargets.map(({ id, name }) => (
+            <EuiFlexItem grow={false} key={id}>
+              <EuiBadge color="hollow">{name}</EuiBadge>
```

Comment:
Great ideas. I didn't refactor `SpaceBadge` but I did add all of the asks above:

<details>
<summary>(expand to see GIF)</summary>

![Jul-23-2020 13-14-39](https://user-images.githubusercontent.com/5295965/88316983-9dc99680-cce6-11ea-9930-6148842e59d3.gif)
</details>

I set the display limit to 5, if there are more than 5 authorized spaces then it will hide any additional behind the expand button (along with any unauthorized).